src/Python/sirf/contrib/NEMA/generate_nema_rois.py

```diff
@@ -0,0 +1,257 @@
+
+def do_registration(recon_image):
+     
+    parfile=pet.get_STIR_examples_dir()+'/samples/stir_math_ITK_output_file_format.par'
+    recon_image.write_par(data_output_path+'recon.nii',parfile)
+    recon_nii=Reg.NiftiImageData3D(data_output_path+'recon.nii')
+    spheres_nii=Reg.NiftiImageData3D(data_output_path+'sphere.nii')
+    sphere_nii= []
+
+    for i in range(1,6+1):
+        sphere_nii.append(Reg.NiftiImageData3D(data_output_path+'sphere'+str(i)+'.nii'))
+
+    #now let's register
+    # The following has a bug so we have used image1.write_par() to create nii images
+    # osem_nii = Reg.NiftiImageData(osem_image)
+    # spheres_nii = Reg.NiftiImageData(image)
+
+    # Set to NiftyF3dSym for non-rigid
+    algo = Reg.NiftyAladinSym()
+
+    # Set images
+    algo.set_reference_image(recon_nii)
+    algo.set_floating_image(spheres_nii)
+    #set parameters
+    algo.set_parameter('SetPerformRigid','1')
+    algo.set_parameter('SetPerformAffine','0')
+    algo.process()
+    reg_image = algo.get_output()
+
+    # reg_image_sirf = Reg.ImageData(reg_image_nii)
+
+    np.set_printoptions(precision=3,suppress=True)
+    TM = algo.get_transformation_matrix_forward()
+    print(TM.as_array())
+    return TM, reg_image, sphere_nii
+
+def generate_nema_rois(recon_image):
+    
+    construct_NEMA_spheres_and_save(acq_data,image_size)
+    TM, reg_image, sphere_nii = do_registration(recon_image)    
+
+    #once we have the registration matrix we can then apply it to the sphere generation
+    resampler = Reg.NiftyResample()
+
+    # Make sure we know what the resampled image domain looks like (this can be the same as the image to resample)
+    resampler.set_reference_image(reg_image)
+    # Add the desired transformation to apply
+    resampler.add_transformation(TM)
+    resampler.set_padding_value(0)
+    # Use linear interpolation
+    resampler.set_interpolation_type_to_nearest_neighbour()
+
+    for j in range(6):
+    # Set image to resample
+        resampler.set_floating_image(sphere_nii[j])
+    # Go!
+        resampler.process()
+        Roi = resampler.get_output()
+        Roi.write(data_output_path+'S'+str(j+1)+'.nii')#TODO getting sirf imagedata to nifty to work without messing the orientation
+        ROIsirf = pet.ImageData(data_output_path+'S'+str(j+1)+'.nii')
+        ROIsirf.write('S'+str(j+1))
+
+if __name__ == '__main__':
+    from docopt import docopt
+    args = docopt(__doc__, version=__version__)
+    #print(args)
+    data_output_path = args['--outpath']
+    if data_output_path is None:
+            data_output_path =  './'
+    prefix = data_output_path + '/'
+
+    sino_file = args['--sino']
+    if sino_file is None:
+        sys.exit('Missing the input sinogram as interfile')
+
+    xy_size = args['--xysize']
+    if xy_size is None:
+        print('Warning: I am setting x and y size to 150 make sure your reconstructed image has the same dimension')
+        xy_size=150
```

Comment:
By using the `docopt` default, this will never happen. Alternatively, you leave the `docopt` string as it currently is such that you can write this message. If so however, then you need "... to 150. Make sure ..."